pkgs/development/python-modules/black/default.nix

```diff
@@ -64,6 +69,9 @@ buildPythonPackage rec {
 
   propagatedBuildInputs = [
     click
+    hatchling
+    hatch-fancy-pypi-readme
+    hatch-vcs
```

Comment:
Those go into nativeBuildInputs. Can we get away with removing hatch-fancy-pypi-readme ?

Reply:
Moved the dependencies to `nativeBuildInputs`.

If we remove `hatch-fancy-pypi-readme` the build fails with

```
Executing pipBuildPhase
Creating a wheel...
WARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
Processing /build/black-22.10.0
  Running command Preparing metadata (pyproject.toml)
  Traceback (most recent call last):
    File "/nix/store/d59d879qx54id3mpmryvx7h8fnndlgna-python3.10-pip-22.2.2/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
      main()
    File "/nix/store/d59d879qx54id3mpmryvx7h8fnndlgna-python3.10-pip-22.2.2/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/nix/store/d59d879qx54id3mpmryvx7h8fnndlgna-python3.10-pip-22.2.2/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 164, in prepare_metadata_for_build_wheel
      return hook(metadata_directory, config_settings)
    File "/nix/store/3zbngm120p9fk301ld3cylyivg3bpc3s-python3.10-hatchling-1.9.0/lib/python3.10/site-packages/hatchling/build.py", line 42, in prepare_metadata_for_build_wheel
      directory = os.path.join(metadata_directory, f'{builder.artifact_project_id}.dist-info')
    File "/nix/store/3zbngm120p9fk301ld3cylyivg3bpc3s-python3.10-hatchling-1.9.0/lib/python3.10/site-packages/hatchling/builders/wheel.py", line 549, in artifact_project_id
      self.project_id
    File "/nix/store/3zbngm120p9fk301ld3cylyivg3bpc3s-python3.10-hatchling-1.9.0/lib/python3.10/site-packages/hatchling/builders/plugin/interface.py", line 355, in project_id
      self.__project_id = f'{self.normalize_file_name_component(self.metadata.core.name)}-{self.metadata.version}'
    File "/nix/store/3zbngm120p9fk301ld3cylyivg3bpc3s-python3.10-hatchling-1.9.0/lib/python3.10/site-packages/hatchling/metadata/core.py", line 117, in core
      metadata_hooks = self.hatch.metadata.hooks
    File "/nix/store/3zbngm120p9fk301ld3cylyivg3bpc3s-python3.10-hatchling-1.9.0/lib/python3.10/site-packages/hatchling/metadata/core.py", line 1366, in hooks
      raise UnknownPluginError(f'Unknown metadata hook: {hook_name}')
  hatchling.plugin.exceptions.UnknownPluginError: Unknown metadata hook: fancy-pypi-readme
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /nix/store/wyhbl43ycqn43d08v5fqj1j6ynf7nz73-python3-3.10.7/bin/python3.10 /nix/store/d59d879qx54id3mpmryvx7h8fnndlgna-python3.10-pip-22.2.2/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py prepare_metadata_for_build_wheel /build/tmppsdbzf8a
  cwd: /build/black-22.10.0
  Preparing metadata (pyproject.toml) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.
```